### PR TITLE
[iOS] - Fix Leo bugs found in QA iteration 1

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary
- Fix popover not showing correctly after tapping on `Advanced Settings
- Fix popover not showing correctly when tapping the `info` icon while the keyboard is on screen
- Update strings to lower-case by default (upper-cased in code)
- Allow QA to be redirected to the release page to manage subscriptions on a release sandboxed build

Resolves https://github.com/brave/brave-browser/issues/36650
Resolves https://github.com/brave/brave-browser/issues/36653
Resolves https://github.com/brave/brave-browser/issues/36656
Resolves https://github.com/brave/brave-browser/issues/36659

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/brave-ios/Sources/AIChat/Subscription/SDK/BraveStoreSDK.swift [hunk 1]
```diff
@@ -159,6 +160,10 @@ public class BraveStoreSDK: AppStoreSDK {
 
   /// The current store environment
   public var enviroment: BraveStoreEnvironment {
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/AIChat/Subscription/SDK/BraveStoreSDK.swift:162]: unrelated typo btw: `enviroment` → `environment`

File: ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift [hunk 2]
```diff
@@ -54,18 +54,43 @@ struct BravePopoverView<Content: View & PopoverContentComponent>: UIViewControll
...
-      context.coordinator.presentedViewController = .init(controller)
-      controller.popoverDidDismiss = { _ in
-        self.isPresented = false
-      }
+      if let parent = uiViewController.parent, !parent.isBeingDismissed {
+        let controller = PopoverController(content: content)
+        context.coordinator.presentedViewController = .init(controller)
+        controller.popoverDidDismiss = { _ in
+          self.isPresented = false
+        }
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift:75]: This is only called when someone calls `dismissPopover` on it btw

File: ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift [hunk 3]
```diff
@@ -54,18 +54,43 @@ struct BravePopoverView<Content: View & PopoverContentComponent>: UIViewControll
...
+      guard uiViewController.presentedViewController == nil
       else {
+        // The system dismissed our popover automatically, but never updated our presentation state
+        // It usually does this if you present another popover or sheet
+        // Manually update it
+        if context.coordinator.presentedViewController != nil {
+          DispatchQueue.main.async {
+            isPresented = false
+          }
+        }
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift:66]: Wouldn't this `updateUIViewController` only be called anyways if the binding/content inside changes? Does this code even hit ever unless you try to present again after its been auto-dismissed once?

Comment (Brandon-T) [reply ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift:66]: It gets called every time something happens. So if another controller gets presented on top of ours, this gets called for some odd reason, even if nothing changes.

**Review by kylehickinson — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]

**Comment by Uni-verse** [comment]
Verified https://github.com/brave/brave-browser/issues/36653 using `iPhone 12` running `iOS 17.3.1` using version `1.65 (77) Nightly channel`

- Ensured that premium is enabled using active credentials
- Ensured that `Manage Subscriptions` sends user to `account.bravesoftware.com` for Nightly channel

https://github.com/brave/brave-core/assets/17885425/205de8fd-2435-43f5-af30-8cf4d06f3f6f

**Comment by hffvld** [comment]
Verified [#36659](https://github.com/brave/brave-browser/issues/36659) on `iPhone 14` using version(s):
```
Device/OS: iPhone 14 / iOS 17.4
Brave build: 1.65 (77)
BraveCore: 1.65.77 (123.0.6312.29)
```
---------------------------

**STEPS:**
1. Followed the steps from https://github.com/brave/brave-browser/issues/36659#issue-2176969502
2. *Verified*

**ACTUAL RESULTS:**
- Verified that the `Regenerate` function works towards each query in the chat.

---------------------------

https://github.com/brave/brave-core/assets/128532965/754aa6fc-3acb-48f7-b083-dcac1dca1056

**Pull request merged**